lib/vagrant-libvirt/templates/domain.xml.erb

```diff
@@ -14,11 +14,7 @@
   <% end %>
 
   <os>
-  <% if @machine_type %>
-    <type machine='<%= @machine_type %>'>hvm</type>
-  <% else %>
-    <type>hvm</type>
-  <% end %>
+    <type <% if @machine_arch %>arch="<%= @machine_arch %>"<% end %> <% if @machine_type %> machine='<%= @machine_type %>'<% end %>>hvm</type>
```

Comment:
simplify please as was before. it should be mainly readable and next step short.